azurelinuxagent/ga/monitor.py

```diff
@@ -416,3 +419,11 @@ def send_cgroup_telemetry(self):
                 logger.warn("Monitor: updating tracked extensions raised {0}: {1}", e, traceback.format_exc())
 
             self.last_cgroup_telemetry = datetime.datetime.utcnow()
+
+    def log_route_table(self):
+        raw_route_list = self.osutil.read_route_table()
+        digest = hash_strings(raw_route_list)
```

Comment:
Is it possible for the route list to be the same, but be in a different order.  The hash will change, but the meaning really hasn't.  I wouldn't think so...

```text
{"a": 1, "b": 2} == {"b": 2, "a": 1} but the hashes are different.
```

Reply:
It's possible, but since /proc/net/route is a direct pipeline to the kernel data structure, there'd be no reason for the order of entries to change if nothing perturbed it. My testing shows the table (as exposed through /proc) doesn't change over a period of hours if there's no actual changes to routes; that's good enough for me.

Reply:
Me too.